upload/extension/opencart/catalog/model/payment/opayo.php

```diff
@@ -0,0 +1,707 @@
+	}
+
+	/**
+	 * Update Subscription Order
+	 *
+	 * @param int    $subscription_id
+	 * @param string $next_payment
+	 *
+	 * @return void
+	 */
+	private function updateSubscriptionOrder(int $subscription_id, string $next_payment): void {
+		$this->db->query("UPDATE `" . DB_PREFIX . "opayo_order_subscription` SET `next_payment` = '" . $next_payment . "', `date_modified` = now() WHERE `subscription_id` = '" . (int)$subscription_id . "'");
+	}
+
+	/**
+	 * Get Subscription Order
+	 *
+	 * @param int $subscription_id
+	 *
+	 * @return array
+	 */
+	private function getSubscriptionOrder(int $subscription_id): array {
+		$query = $this->db->query("SELECT * FROM `" . DB_PREFIX . "opayo_order_subscription` WHERE `subscription_id` = '" . (int)$subscription_id . "'");
+
+		return $query->row;
+	}
+
+	/**
+	 * Add Subscription Transaction
+	 *
+	 * @param int   $subscription_id
+	 * @param int   $order_id
+	 * @param array $response_data
+	 * @param int   $type
+	 *
+	 * @return void
+	 */
+	private function addSubscriptionTransaction(int $subscription_id, int $order_id, array $response_data, int $type): void {
+		$this->db->query("INSERT INTO `" . DB_PREFIX . "subscription_transaction` SET `subscription_id` = '" . (int)$subscription_id . "', `order_id` = '" . (int)$order_id . "', `date_added` = NOW(), `amount` = '" . (float)$response_data['Amount'] . "', `type` = '" . (int)$type . "', `reference` = '" . $this->db->escape($response_data['VendorTxCode']) . "'");
+	}
+
+	/**
+	 * Get Profiles
+	 *
+	 * @return array
+	 */
+	private function getProfiles(): array {
+		$query = $this->db->query("SELECT `or`.`order_recurring_id` FROM `" . DB_PREFIX . "order_recurring` `or` JOIN `" . DB_PREFIX . "order` `o` USING(`order_id`) WHERE `o`.`payment_code` = 'opayo'");
+
+		$subscriptions = [];
+
+		foreach ($query->rows as $subscription) {
+			$subscriptions[] = $this->getProfile($subscription['subscription_id']);
+		}
+
+		return $subscriptions;
+	}
+
+	/**
+	 * Get Profile
+	 *
+	 * @param int $subscription_id
+	 *
+	 * @return array
+	 */
+	private function getProfile(int $subscription_id): array {
+		$query = $this->db->query("SELECT * FROM `" . DB_PREFIX . "subscription` WHERE `subscription_id` = " . (int)$subscription_id);
+
+		return $query->row;
+	}
+
+	/**
+	 * Update Cron Run Time
+	 *
+	 * @return void
+	 */
+	public function updateCronRunTime(): void {
+		$this->db->query("DELETE FROM `" . DB_PREFIX . "setting` WHERE `code` = 'opayo' AND `key` = 'payment_opayo_last_cron_run'");
+		$this->db->query("INSERT INTO `" . DB_PREFIX . "setting` (`store_id`, `code`, `key`, `value`, `serialized`) VALUES (0, 'opayo', 'payment_opayo_last_cron_run', NOW(), 0)");
+	}
+
+	/**
+	 * Send Curl
+	 *
+	 * @param string $url
+	 * @param array  $payment_data
```

Comment:
Missing value type for array:
```suggestion
	 * @param array<mixed>  $payment_data
```